qiskit_experiments/library/characterization/t2ramsey.py

```diff
@@ -116,7 +117,8 @@ def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
             circ = qiskit.QuantumCircuit(1, 1)
             circ.h(0)
             circ.delay(delay, 0, self._unit)
-            circ.p(2 * np.pi * self._osc_freq, 0)
+            rotation_angle = 2 * np.pi * self._osc_freq * delay
```

Comment:
This is correct only if `delay` is in seconds. We should convert to seconds to ensure that is true.

Reply:
I think the assumption they here is that `delay` and `osc_freq` and are same units (inverse units) so these should cancel out ie if units=s freq is Hz, but if units = us, freq is MHz

Reply:
@merav-aharoni This assumption would break though if `units = 'dt'` so maybe best to just always make osc_freq Hz like @blakejohnson suggests